patches/tModLoader/Terraria/Initializers/LaunchInitializer.cs.patch

```diff
@@ -22,6 +22,59 @@
  		}
  
  		private static void LoadClientParameters(Main game) {
+-			string iP;
+-			if ((iP = TryParameter("-j", "-join")) != null)
+-				game.AutoJoin(iP);
++			string ip;
++			if ((ip = TryParameter("-j", "-join")) != null) {
++
++				int port = -1;
++				if (ip.Contains(':')) {
++					int colonInd = ip.LastIndexOf(':');
++					if (int.TryParse(ip.Substring(colonInd + 1), out int portNum)) {
++						port = portNum;
++						ip = ip.Substring(0, colonInd);
++					}
++				}
++
++				if (!Netplay.SetRemoteIP(ip)) {
++					ModLoader.Logging.tML.Error("Could not set remote IP: " + ip);
++				}
++				else {
++					if (port >= 0)
++						Netplay.ListenPort = port;
++
++					ModLoader.ModLoader.OnSuccessfulLoad += () => {
++						string player = TryParameter("-plr", "-player");
```

Comment:
If `-plr` or `-player` is not supplied we should keep the vanilla behaviour and go to the player select menu.

Reply:
That's what it does already